### PR TITLE
YC-1143 - Bypass max submissions

### PR DESCRIPTION
**Pull request by danickfort**

### CODE REVIEW

File: geocity/apps/forms/migrations/0027_rename_max_submissions_bypass_authorized_form_max_submissions_bypass_enabled.py
Comment (rbovard) [inline geocity/apps/forms/migrations/0027_rename_max_submissions_bypass_authorized_form_max_submissions_bypass_enabled.py:1]: Is possible to squash the migrations?

File: geocity/apps/forms/migrations/0026_form_max_submissions_bypass_authorized.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        ("forms", "0025_alter_form_quick_access_slug"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="form",
+            name="max_submissions_bypass_authorized",
+            field=models.BooleanField(
+                default=False,
+                verbose_name="Autoriser les utilisateurs avec permission à soumettre des demandes même si le nombre maximal est atteint",
```
Comment (rbovard) [inline geocity/apps/forms/migrations/0026_form_max_submissions_bypass_authorized.py:18]:
```suggestion
                verbose_name="Autoriser le secrétariat à soumettre des demandes même si le nombre maximal est atteint",
```

File: geocity/apps/forms/models.py [hunk 2]
```diff
@@ -501,6 +501,12 @@ class Form(models.Model):
         null=True,
         blank=True,
     )
+    max_submissions_bypass_enabled = models.BooleanField(
+        _(
+            "Autoriser les utilisateurs avec permission à soumettre des demandes même si le nombre maximal est atteint"
```
Comment (rbovard) [inline geocity/apps/forms/models.py:506]:
```suggestion
            "Autoriser le secrétariat à soumettre des demandes même si le nombre maximal est atteint"
```

File: geocity/apps/submissions/forms.py [hunk 3]
```diff
@@ -272,8 +273,15 @@ def __init__(self, instance, form_categories=None, *args, **kwargs):
...
                 form_name = form.name
                 if form.has_exceeded_maximum_submissions():
-                    form_name = f"{form_name} <span class='px-5 text-danger'>{form.max_submissions_message}</span>"
-                    disabled_choices.add(form.pk)
+                    max_submission_msg = f"<span class='px-5 text-danger'>{form.max_submissions_message}</span>"
+                    if (
+                        form.max_submissions_bypass_enabled
+                        and has_permission_to_amend_submission(self.user, self.instance)
+                    ):
+                        form_name = f"{form_name} <s>{max_submission_msg}</s> <span class='px-5 text-danger'>(formulaire actif pour l'utilisateur courant)</span>"
```
Comment (rbovard) [inline geocity/apps/submissions/forms.py:281]:
```suggestion
                        form_name = f"{form_name} <s class="text-danger">{max_submission_msg}</s> <span class='text-danger'>(formulaire actif pour l'utilisateur courant)</span>"
```

**Review by rbovard — CHANGES_REQUESTED** [review]
Works fine, thanks a lot.

I've some remarks and made some code suggestions

File: geocity/apps/submissions/forms.py [hunk 4]
```diff
@@ -272,8 +273,15 @@ def __init__(self, instance, form_categories=None, *args, **kwargs):
...
                 form_name = form.name
                 if form.has_exceeded_maximum_submissions():
-                    form_name = f"{form_name} <span class='px-5 text-danger'>{form.max_submissions_message}</span>"
-                    disabled_choices.add(form.pk)
+                    max_submission_msg = f"<span class='pl-3 text-danger'>{form.max_submissions_message}</span>"
+                    if (
+                        form.max_submissions_bypass_enabled
+                        and has_permission_to_amend_submission(self.user, self.instance)
+                    ):
+                        form_name = f"{form_name} <s class='text-danger'>{max_submission_msg}</s> <span class='px-5 text-danger'>(formulaire actif pour l'utilisateur courant)</span>"
```
Comment (rbovard) [inline geocity/apps/submissions/forms.py:281]:
```suggestion
                        form_name = f"{form_name} <s class='text-danger'>{max_submission_msg}</s> <span class='pl-1 text-danger'>(formulaire actif pour l'utilisateur courant)</span>"
```

**Review by rbovard — APPROVED** [review]
Thanks a lot!

(I've just done a little style adaptation)

**Comment by rbovard** [comment]
@danickfort Could you please just check the conflict due to the merge of #877? I think it's pretty straightforward but I prefer to let you do that. After that you can merge.

**Comment by danickfort** [comment]
> @danickfort Could you please just check the conflict due to the merge of #877? I think it's pretty straightforward but I prefer to let you do that. After that you can merge (edit: after the tests are green ;)).

Done 👍

**Pull request merged**